src/main/java/com/xwiki/authentication/saml/samlauth/SamlAuthenticator.java

```diff
@@ -0,0 +1,108 @@
+import com.onelogin.saml2.settings.SettingsBuilder;
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.user.api.XWikiUser;
+import com.xwiki.authentication.saml.AuthenticatedUserHandler;
+import com.xwiki.authentication.saml.NonAuthenticatedAccessHandler;
+import com.xwiki.authentication.saml.SamlAuthenticationHandler;
+import com.xwiki.authentication.saml.xwiki.XWikiGroupManager;
+import com.xwiki.authentication.saml.function.SupplierWithException;
+import com.xwiki.authentication.saml.onelogin.OneLoginAuth;
+import java.util.Optional;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+
+public class SamlAuthenticator {
+    public static final EntityReference PROFILE_PARENT = new EntityReference("XWikiUsers", EntityType.DOCUMENT,
+            new EntityReference(XWiki.SYSTEM_SPACE, EntityType.SPACE));
+    private final Saml2Settings samlSettings;
+    private final SamlAuthConfig authConfig;
+    private final DocumentReferenceResolver<String> currentMixedDocumentReferenceResolver;
+    private final EntityReferenceSerializer<String> compactStringEntityReferenceSerializer;
+    private final OneLoginAuth loginAuthFactory;
+
+    private final XWikiGroupManager groupManager;
+
+    public SamlAuthenticator(SamlAuthConfig authConfig,
+                             DocumentReferenceResolver<String> currentMixedDocumentReferenceResolver,
+                             EntityReferenceSerializer<String> compactStringEntityReferenceSerializer,
+                             OneLoginAuth loginAuthFactory,
+                             XWikiGroupManager groupManager) {
+        this.groupManager = groupManager;
+        this.authConfig = authConfig;
+        this.currentMixedDocumentReferenceResolver = currentMixedDocumentReferenceResolver;
+        this.compactStringEntityReferenceSerializer = compactStringEntityReferenceSerializer;
+        this.loginAuthFactory = loginAuthFactory;
+        this.samlSettings = buildSamlSettings();
+    }
+
+    private Saml2Settings buildSamlSettings() {
+        final java.util.Properties settings = new java.util.Properties();
+        settings.put("onelogin.saml2.strict", true);
+        settings.put("onelogin.saml2.sp.entityid", authConfig.entityId);
+        settings.put("onelogin.saml2.sp.assertion_consumer_service.url", authConfig.assertionConsumerServiceUrl);
+        settings.put("onelogin.saml2.sp.nameidformat", authConfig.nameIdFormat);
+        settings.put("onelogin.saml2.idp.entityid", authConfig.idpEntityId);
+        settings.put("onelogin.saml2.idp.single_sign_on_service.url", authConfig.idpSingleSignOnUrl);
+        settings.put("onelogin.saml2.idp.x509cert", authConfig.x509Certificate);
+        return new SettingsBuilder().fromProperties(settings).build();
+    }
```

Comment:
Using a Map seems more appropriate than properties:
```suggestion
    private Saml2Settings buildSamlSettings() {
        final HashMap<String, Object> settings = new HashMap<>();
        settings.put("onelogin.saml2.strict", true);
        settings.put("onelogin.saml2.sp.entityid", authConfig.entityId);
        settings.put("onelogin.saml2.sp.assertion_consumer_service.url", authConfig.assertionConsumerServiceUrl);
        settings.put("onelogin.saml2.sp.nameidformat", authConfig.nameIdFormat);
        settings.put("onelogin.saml2.idp.entityid", authConfig.idpEntityId);
        settings.put("onelogin.saml2.idp.single_sign_on_service.url", authConfig.idpSingleSignOnUrl);
        settings.put("onelogin.saml2.idp.x509cert", authConfig.x509Certificate);
        return new SettingsBuilder().fromValues(settings).build();
    }
```